test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/StrictCheckSpanProcessor.java

```diff
@@ -27,58 +26,39 @@ public StrictCheckSpanProcessor() {
 
     }
 
+    private static Map<String, MockSpanData> finishedSpanItems = new ConcurrentHashMap<>();
+
     @Override
     public void onStart(Span span) {
         spanMap.put(span.getSpanId(), Thread.currentThread().getStackTrace());
+        MockSpanData spanData;
+        if (span.getParentSpan() == null){
+            spanData = new MockSpanData(span.getSpanId(),
+                "0000000000", span.getTraceId(), System.nanoTime(), false, span.getSpanName());
+        } else {
+            spanData = new MockSpanData(span.getSpanId(),
+                span.getParentSpan().getSpanId(), span.getTraceId(), System.nanoTime(), false, span.getSpanName());
+        }
```

Comment:
1. We should extract to method say `toMockSpanData`
2. Why are we updating parent?